libxrdp/xrdp_orders_rail.c

```diff
@@ -260,6 +266,12 @@ xrdp_orders_send_window_new_update(struct xrdp_orders *self, int window_id,
     {
         /* titleInfo */
         num_chars = g_mbstowcs(0, window_state->title_info, 0);
+        num_chars = MIN(num_chars, sizeof(twchar) * (g_strlen(window_state->title_info) + 1));
```

Comment:
What is happening here? I think g_mbstowcs() should be fully trusted, as it returns the number of the characters. But we should add 1 for the terminating zero and multiply the result by `sizeof(twchar)` to get the number of bytes.

g_strlen() can overestimate the number of the wide characters, as it would count multi-byte characters as multiple characters.